lib/ops/opsservice/agents.go

```diff
@@ -457,10 +457,6 @@ func (r *AgentPeerStore) checkHostname(ctx context.Context, group *agentGroup, a
 		return trace.AccessDenied("One of existing servers already has hostname %q: %q.", hostname, existingServers)
 	}
 
-	if group.hasPeer(addr, hostname) {
```

Comment:
I've removed this test to account for the case when the joining agent fails with an error loosing the connection to the cluster. It should be able to reconnect (rejoin) at the expense of the unlikelier case of having two agents starting from the same node. Let me know if you still think this is an omission.

Reply:
IIRC it wasn't for the case of starting two agents from the same node (which I agree is unlikely) but rather to catch the case when two different nodes have the same hostname (which is very likely and we've run into this in the past).

Reply:
Oh - my bad. I rechecked the condition and you're right. Will revert.